storage/remote/queue_manager.go

```diff
@@ -1366,6 +1420,7 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue *queue) {
 		pBufRaw []byte
 		buf     []byte
 	)
+	// TODO(@tpaschalis) Should we also raise the max if we have WAL metadata?
```

Comment:
I think no, and we can remove this todo. Metadata will be just referenced for each time series in the request. It's not an additional peudo-sample type like exemplars.

We should double check that we're not counting 'pending metadata' against the max when appending data